### ISSUE TITLE
"http://ftp.drupal.org/files/projects/drupal-7.39.zip" file could not be downloaded: Peer certificate CN=`j.ssl.fastly.net' did not match expected CN=`ftp.drupal.org' 

### ISSUE THREAD

**Issue by mathieuhelie** [composer/composer#4818]
After upgrading from december to january version of composer our custom package to install drupal from 
```
"dist": {
        "url": "http://ftp.drupal.org/files/projects/drupal-7.39.zip",
        "type": "zip"
      },
```
fails to install or update with the following error.
```
 [Composer\Downloader\TransportException]                                                                                                            
  The "http://ftp.drupal.org/files/projects/drupal-7.39.zip" file could not be downloaded: Peer certificate CN=`j.ssl.fastly.net' did not match expe  
  cted CN=`ftp.drupal.org'                                                                                                                            
  Failed to enable crypto                                                                                                                             
  failed to open stream: operation failed      
```

Drupal.org is now provided through the Fastly varnish cloud.

Downgrading composer to 1.0.0-alpha11 resolves the issue.

`PHP 5.5.9-1ubuntu4.14 (cli) (built: Oct 28 2015 01:34:46) `

**Comment by Tuurlijk** [comment]
On Ubuntu:

```
Installing dependencies
- Installing typo3/cms-composer-installers (1.1.4) Downloading
- Installing helhum/class-alias-loader (1.1.5) Downloading
- Installing doctrine/instantiator (1.0.5) Downloading

[RuntimeException] The \"https://get.typo3.org/json\" file could not be downloaded: Peer certificate CN=`ssl340684.cloudflaressl.com' did not match expected CN=`get.typo3.org' 
Failed to enable crypto failed to open stream: operation failed [Composer\\Downloader\\TransportException] 
The \"https://get.typo3.org/json\" file could not be downloaded: Peer certificate CN=`ssl340684.cloudflaressl.com' did not match expected CN=`get.typo3.org' 
Failed to enable crypto failed to open stream: operation failed install
```

Get.typo3.org has a SNI Cert.

This broke in the last week or so for us.

**Comment by jakoch** [comment]
That's again the SSL forwarding bug with subjectAltName, #4782.
Bugfix ongoing: #4798.

Meanwhile, you might bypass this issue by rolling back to a previous version of composer:
`curl -sS https://getcomposer.org/installer | php -- --version=1.0.0-alpha10`

**Comment by webflo** [comment]
@mathieuhelie Drupal-Packagist switched to two https urls on friday, this might fix your problem.

**Comment by alcohol** [comment]
Closing duplicate of #4782

**Comment by Seldaek** [comment]
This wasn't really a duplicate, but anyway both that issue and this one should now be fixed, please try again after running `composer self-update`!

**Comment by alcohol** [comment]
@Seldaek different domain same issue..?

**Comment by Seldaek** [comment]
Nope, this issue was about SAN and not redirects. Fixed by #4798

**Comment by alcohol** [comment]
My bad.
